kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/bootstrap/FilterChainFactory.java

```diff
@@ -64,9 +64,8 @@ public Filter create(FilterFactoryContext context) {
 
         @Override
         public void close() {
-            if (!closed) {
-                closed = true;
-                filterFactory.close();
+            if (!this.closed.getAndSet(true)) {
```

Comment:
nit:
```suggestion
            if (this.closed.compareAndSet(false, true)) {
```
Is maybe a little easier to read.

Reply:
Personally I find `compareAndSet` a bit _harder_ to read because:

1. I'm never completely sure about the order of the arguments, so have to consult the Javadocs
2. It's a bit like writing `if (closed == false)`, which it not code I'd ever write, since ` if (!closed)` is much clearer.

Reply:
I'm happy to leave it as is, but I say its easier to read because it only executes the code in the if block if the transition from X to Y was successful which improves the readability of the whole construct not just the specific invocation.